app/services/imports/case_logs_import_service.rb

```diff
@@ -0,0 +1,278 @@
+      local_authority: 2,
+    }.freeze
+
+    IRPRODUCT = {
+      rent_to_buy: 1,
+      london_living_rent: 2,
+      other_intermediate_rent_product: 3,
+    }.freeze
+
+    # These must match our form
+    RENT_TYPE = {
+      social_rent: 0,
+      affordable_rent: 1,
+      london_affordable_rent: 2,
+      rent_to_buy: 3,
+      london_living_rent: 4,
+      other_intermediate_rent_product: 5,
+    }.freeze
+
+    # Order matters since we derive from previous values (uses attributes)
+    def create_log(xml_doc)
+      attributes = {}
+
+      # Required fields for status complete or logic to work
+      attributes["startdate"] = start_date(xml_doc)
+      attributes["owning_organisation_id"] = find_organisation_id(xml_doc, "OWNINGORGID")
+      attributes["managing_organisation_id"] = find_organisation_id(xml_doc, "MANINGORGID")
+      attributes["previous_postcode_known"] = previous_postcode_known(xml_doc)
+      attributes["ppostcode_full"] = previous_postcode(xml_doc, attributes)
+      attributes["needstype"] = needs_type(xml_doc)
+      attributes["lar"] = london_affordable_rent(xml_doc)
+      attributes["irproduct"] = unsafe_string_as_integer(xml_doc, "IRPRODUCT")
+      attributes["irproduct_other"] = field_value(xml_doc, "xmlns", "IRPRODUCTOTHER")
+      attributes["rent_type"] = rent_type(xml_doc, attributes)
+      attributes["rsnvac"] = unsafe_string_as_integer(xml_doc, "Q27")
+      attributes["renewal"] = renewal(attributes)
+      (1..8).each do |index|
+        attributes["age#{index}"] = age(xml_doc, index)
+        attributes["sex#{index}"] = sex(xml_doc, index)
+        attributes["ecstat#{index}"] = unsafe_string_as_integer(xml_doc, "P#{index}Eco")
+      end
+      # attributes["hhmemb"] =
+
+      # Not specific to our form but required for CDS and can't be inferred
+      # attributes["form"] = Integer(field_value(xml_doc, "xmlns", "FORM"))
+      # attributes["lettype"] = let_type(xml_doc, attributes)
+
+      case_log = CaseLog.new(attributes)
+      case_log.save!
+
+      # pp attributes
+      # pp case_log.status
+      # pp case_log.send(:mandatory_fields)
+    end
+
+    def start_date(xml_doc)
+      day = Integer(field_value(xml_doc, "xmlns", "DAY"))
+      month = Integer(field_value(xml_doc, "xmlns", "MONTH"))
+      year = Integer(field_value(xml_doc, "xmlns", "YEAR"))
+      Date.new(year, month, day)
+    end
+
+    def get_form_name_component(xml_doc, index)
+      form_name = field_value(xml_doc, "meta", "form-name")
+      form_type_components = form_name.split("-")
+      form_type_components[index]
+    end
+
+    def needs_type(xml_doc)
+      gn_sh = get_form_name_component(xml_doc, -1)
+      case gn_sh
+      when "GN"
+        GN_SH[:general_needs]
+      when "SH"
+        GN_SH[:supported_housing]
+      else
+        raise "Unknown needstype value: #{gn_sh}"
+      end
+    end
+
+    # This does not match renttype (CDS) which is derived by case logs logic
```

Comment:
Is this comment still true?

Reply:
Yes, our rent_type won't match what CDS has.

Reply:
Isn't that a problem?

Reply:
Nothing we can do, this is one field that needs to be resolved at export. The way we capture the rent type is different (we don't rely on let_type / lar etc...)